src/Controller/EnMarche/CommitteeController.php

```diff
@@ -48,13 +50,80 @@ public function showAction(Request $request, Committee $committee): Response
 
         $committeeManager = $this->get('app.committee.manager');
 
-        return $this->render('committee/show.html.twig', [
-            'committee' => $committee,
-            'committee_hosts' => $committeeManager->getCommitteeHosts($committee),
-            'committee_timeline' => $committeeManager->getTimeline($committee, $this->getParameter('timeline_max_messages')),
-            'committee_timeline_max_messages' => $this->getParameter('timeline_max_messages'),
-            'form' => $form ? $form->createView() : null,
-        ]);
+        return $this->render('committee/show.html.twig',
+            [
+                'committee' => $committee,
+                'committee_hosts' => $committeeManager->getCommitteeHosts($committee),
+                'committee_timeline' => $committeeManager->getTimeline($committee,
+                    $this->getParameter('timeline_max_messages')),
+                'committee_timeline_max_messages' => $this->getParameter('timeline_max_messages'),
+                'form' => $form ? $form->createView() : null,
+            ]);
+    }
+
+    /**
+     * @Route("/timeline/{id}/edit", name="app_committee_timeline_edit")
+     * @ParamConverter("committee", options={"mapping":{"slug": "slug"}})
+     * @ParamConverter("committeeFeedItem", options={"mapping":{"id": "id"}})
+     * @Method("GET|POST")
+     * @Security("is_granted('SHOW_COMMITTEE', committee)")
+     */
+    public function timelineEditAction(Request $request, Committee $committee, CommitteeFeedItem $committeeFeedItem): Response
+    {
+        if (!($user = $this->getUser()) instanceof Adherent
```

Comment:
What about using a Voter with this logic ? 
And you can use for example : `@Security("is_granted('EDIT_TIMELINE', committee)")`